.github/workflows/book.yml

```diff
@@ -0,0 +1,38 @@
+name: deploy-book
+
+# Only run this when the master branch changes
+on:
+  push:
+    paths:
+      - 'docs/**'
+    branches:
+      - 'master'
+
+# This job installs dependencies, build the book, and pushes it to `gh-pages`
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
```

Comment:
For my education, is this the container being used?

Is it normal for this kind of dependency to arise in a cloud native environment? Is this inherited from using an upstream project and it's not pragmatic to move everything to a standard container environment?

Or am I completely on the wrong rabbit trail here?

Reply:
This specifies the virtual environment for the github action. You can read more about these here: https://github.com/actions/virtual-environments